lib/auth/usertoken.go

```diff
@@ -139,8 +139,14 @@ func (a *Server) CreateResetPasswordToken(ctx context.Context, req CreateUserTok
 		return nil, trace.BadParameter("invalid reset password token request type")
 	}
 
-	_, err = a.ResetPassword(req.Name)
-	if err != nil {
+	switch user, err := a.GetUser(req.Name, false /* withSecrets */); {
+	case err != nil:
+		return nil, trace.Wrap(err)
+	case user.GetUserType() != types.UserTypeLocal:
+		return nil, trace.AccessDenied("only local users may be reset")
+	}
+
+	if _, err := a.ResetPassword(req.Name); err != nil {
```

Comment:
We also got a conflict here, as ResetPassword doesn't take a context in older branches.

Other signature changes are solved in a separate conflict, as the lines didn't conflict on cherry-pick.